envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RateLimitFilterFactoryTest.kt

```diff
@@ -0,0 +1,108 @@
+package pl.allegro.tech.servicemesh.envoycontrol.snapshot.resource.listeners.filters
+
+import io.envoyproxy.envoy.extensions.filters.http.lua.v3.Lua
+import io.envoyproxy.envoy.extensions.filters.http.ratelimit.v3.RateLimit
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import pl.allegro.tech.servicemesh.envoycontrol.groups.CommunicationMode
+import pl.allegro.tech.servicemesh.envoycontrol.groups.Group
+import pl.allegro.tech.servicemesh.envoycontrol.groups.Incoming
+import pl.allegro.tech.servicemesh.envoycontrol.groups.IncomingRateLimitEndpoint
+import pl.allegro.tech.servicemesh.envoycontrol.groups.ProxySettings
+import pl.allegro.tech.servicemesh.envoycontrol.groups.ServicesGroup
+import pl.allegro.tech.servicemesh.envoycontrol.snapshot.RateLimitProperties
+
+class RateLimitFilterFactoryTest {
+    private val properties = RateLimitProperties()
+    private val rateLimitFilterFactory = RateLimitFilterFactory(properties)
+
+    @Test
+    fun `should not create ratelimit filter if no rate limits exist`() {
+        // given
+        val group = createGroupWithIncomingRateLimits()
+
+        // when
+        val filter = rateLimitFilterFactory.createGlobalRateLimitFilter(group)
+
+        // then
+        assertThat(filter).isNull()
+    }
+
+    @Test
+    fun `should create ratelimit filter with proper domain and ratelimit service name`() {
+        // given
+        val group = createGroupWithIncomingRateLimits(
+            IncomingRateLimitEndpoint("/hello", rateLimit = "0/s")
```

Comment:
should it be allowed to setup rate limit to 0?

Reply:
Yes. It basically means it's banned.